### DISCUSSION TITLE
Getting _event.origin in guard?

### DISCUSSION THREAD

**Discussion by rajansaini691** [statelyai/xstate#1730] [Q&A] [answered]
Hello! For my application, I want to use a machine as a kind of mux that routes messages between different services based on its state. To do this, I am trying to combine forwardTo with a guard that depends on the event's origin. The problem is, the SCXML event doesn't appear to have an origin attribute (which is different from what is documented). Should this be expected, and how should one do this kind of conditional routing instead? 

Example code (sorry about line numbers):
```
    4     guards: {
>>  3       localEvt: (context: any, event: any, meta) => {
    2         
    1         
  55          console.log(meta._event)    // Has no origin attribute
    1         return meta._event.origin == "midi";   // fails
    2       }
    3     }

```

**Comment by davidkpiano** [comment, marked as answer]
That's strange, it should. Can you make a reproducible example? [CodeSandbox (XState template)](https://codesandbox.io/s/xstate-example-template-qcwed)

**Reply by rajansaini691** [reply]
[~~Here it is~~](https://codesandbox.io/s/xstate-example-template-vanilla-forked-9dkz6?file=/src/index.js)

~~For some reason in my personal project _event was a child of meta, while this repro matches the docs in that it directly stored the event data. Regardless, origin is undefined~~

**Reply by rajansaini691** [reply]
Also, if you're okay with it, I have another unrelated question: what's the best way use the negation of a guard? I must have missed it in the docs.

**Reply by rajansaini691** [reply]
Sorry, please disregard the previous repro; use [this one](https://codesandbox.io/s/xstate-example-template-vanilla-forked-vkl4t?file=/src/index.js) instead.

**Reply by davidkpiano** [reply]
> what's the best way use the negation of a guard? 

For now, you have to create a new guard that's the negation of that guard.

In version 5, you will be able to do this:

```js
// Tentative API: version 5! Not available in version 4
import { not } from 'xstate/guards';

// ...
on: {
  SOME_EVENT: {
    target: 'somewhere',
    guard: not('someExistingTarget')
  }
}
```

**Reply by rajansaini691** [reply]
Thanks! Would you mind taking a look at the reproduction above?

Also, that's very cool; excited for version 5!

**Reply by Andarist** [reply]
@rajansaini691 I've prepared a fix for this: https://github.com/davidkpiano/xstate/pull/1766

**Reply by rajansaini691** [reply]
@Andarist Thanks so much!
